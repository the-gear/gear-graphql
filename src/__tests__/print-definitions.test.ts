import { readFileSync } from 'fs';
import { parse } from 'graphql';
import { resolve } from 'path';
import printDefinitions from '../print-definitions';

const source = readFileSync(resolve(__dirname, './complex.gql'), 'utf8');

describe('print-kind', () => {
  it('works', () => {
    const document = parse(source);
    const printed = printDefinitions(document);
    expect(printed).toMatchSnapshot();
  });
});
