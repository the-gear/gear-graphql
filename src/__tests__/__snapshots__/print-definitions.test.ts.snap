// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print-kind works 1`] = `
"scalar               URL                              * GraphQL request:2:1
directive            ref                              * GraphQL request:4:1
scalar               Complex                          * GraphQL request:7:1
input                LoginInput                       * GraphQL request:9:1
interface            Node                             * GraphQL request:14:1
type                 LoginResult                      * GraphQL request:18:1
type                 Query                            * GraphQL request:22:1
type                 Mutation                         * GraphQL request:27:1
enum                 Decision                         * GraphQL request:31:1
extend enum          Decision                         * GraphQL request:32:1
extend interface     Node                             * GraphQL request:34:1
extend type          Query                            * GraphQL request:39:1
extend input         LoginInput                       * GraphQL request:43:1
schema                                                * GraphQL request:47:1
type                 Subscription                     * GraphQL request:52:1
extend schema                                         * GraphQL request:56:1"
`;
